### PR TITLE
feat(java): migrate to adoptopenjdk and drop jdk 10 support

### LINKED ISSUE #169: Java buildbox : move to adoptopenjdk instead of openjdk
Because openjdk docker image doesn't contains latest fix (https://mail.openjdk.java.net/pipermail/jdk8u-dev/2019-May/009330.html), we should move to adoptopenjdk images . ...

### PR DESCRIPTION
**Pull request by Crow-EH**
Closes #169

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,9 +17,8 @@ matrix:
     - env: LANGUAGE=dind-aws     VERSION=1        DOCKER_COMPOSE_VERSION=1.24.0 GLIBC_VERSION=2.27-r0 PIP_VERSION=18.1 PIPENV_VERSION=2018.10.13
     - env: LANGUAGE=golang       VERSION=1.9      GLIDE_VERSION=v0.13.1
     - env: LANGUAGE=java         VERSION=6        MAVEN_VERSION=3.2.1
-    - env: LANGUAGE=java         VERSION=8        JAVA_VERSION=8u212-jdk-slim
-    - env: LANGUAGE=java         VERSION=10       JAVA_VERSION=10.0.2-jdk-slim
-    - env: LANGUAGE=java         VERSION=11       JAVA_VERSION=11.0.3-jdk-slim
+    - env: LANGUAGE=java         VERSION=8        JAVA_VERSION=jdk8u212-b04-debian-slim
```
Comment (fpavageau) [inline .travis.yml]: I think we need `JAVA_MAJOR_VERSION` as well?

File: README.md [hunk 2]
```diff
@@ -96,4 +96,4 @@ The `travis.py` script try to be clever:
 
 It is possible to build local image for testing with the following command:
 
-    CI_HELPER_VERSION=0.0.5 MODD_VERSION=0.5 JAVA_VERSION=10.0.1-jdk-slim VERSION=10 python travis.py --language java --pull-request=true
+    CI_HELPER_VERSION=0.0.6 MODD_VERSION=0.5 JAVA_VERSION=jdk-11.0.3_7-debian-slim VERSION=10 python travis.py --language java --pull-request=true
```
Comment (fpavageau) [inline README.md]: The command line is also missing `JAVA_MAJOR_VERSION`, and `VERSION=10` should be replaced by `VERSION=11`.

Comment (Crow-EH) [reply README.md]: Yeah I forgot to update `VERSION`. :+1:

`JAVA_MAJOR_VERSION` is in fact declared in `travis.py` and initlialized from `VERSION` (I didn't add it, it was already done).

Comment (Crow-EH) [reply README.md]: I'll clean it a bit too, we need it for the sed (`Dockerfile.tpl` to `Dockerfile`), but not for the Docker env anymore (it was used to do JDK 8 specific tasks)

File: .travis.yml (hunk 1 above)
Comment (Crow-EH) [reply .travis.yml]: `JAVA_MAJOR_VERSION` is in fact declared in `travis.py` and initlialized from `VERSION` (I didn't add it, it was already done).

Comment (Crow-EH) [reply .travis.yml]: I'll stop using JAVA_MAJOR_VERSION and use directly VERSION :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,8 @@ Versions
 * Upgrade Xdebug version: 2.7.2
 * Upgrade Ruby version: 2.6.3
 * Fix Java 8/11 image issues from debian:stretch-slim
+* Drop JDK 10 support
+* Migrate Java buildboxes to adoptopenjk
```
Comment (mremi) [inline CHANGELOG.md]: `adoptopenjdk`

File: README.md (hunk 2 above)
Comment (Crow-EH) [reply README.md]: The example has been fixed

File: .travis.yml (hunk 1 above)
Comment (Crow-EH) [reply .travis.yml]: `JAVA_MAJOR_VERSION` has been renamed in `VERSION` in `travis.py` & `Dockerfile.tpl`

File: CHANGELOG.md (hunk 3 above)
Comment (Crow-EH) [reply CHANGELOG.md]: Thank you. Fixed.

Comment (fpavageau) [reply CHANGELOG.md]: Or even `AdoptOpenJDK` 😅

**Pull request merged**